nixos/modules/config/system-path.nix

```
@@ -8,8 +8,18 @@
 }:
 let
 
+  requiredPackagesPriority = config.environment.requiredPackagesPriority;
   requiredPackages =
-    map (pkg: lib.setPrio ((pkg.meta.priority or lib.meta.defaultPriority) + 3) pkg)
+    map
+      (
+        pkg:
+        lib.setPrio (
+          if requiredPackagesPriority == null then
+            ((pkg.meta.priority or lib.meta.defaultPriority) + 3)
+          else
+            requiredPackagesPriority
```

Could be squashed into
```suggestion
          config.environment.requiredPackagesPriority or
            ((pkg.meta.priority or lib.meta.defaultPriority) + 3)
```

---

https://nix.dev/manual/nix/2.28/language/operators#attribute-selection

But:
1. only works with full attrpaths, can't be used for single variables
2. I MISUNDERSTOOD, it only check for defined/undefined, not null/non-null or truthy/falsy. Ignore my suggestion, I apologize for misinformation

---

That's alright :)
I assume this thread is resolved?